src/downshift.js

```diff
@@ -668,8 +672,10 @@ class Downshift extends Component {
   }
 
   input_handleKeyDown = event => {
-    if (event.key && this.keyDownHandlers[event.key]) {
-      this.keyDownHandlers[event.key].call(this, event)
+    const handler =
+      this.props.keyDownHandlers[event.key] || this.keyDownHandlers[event.key]
+    if (handler) {
+      handler.call(this.getStateAndHelpers(), event)
```

Comment:
> [From my understanding, this would never differ here - the object returned from getStateAndHelpers() is always bound.](https://github.com/paypal/downshift/pull/346#discussion_r170706095)

Ah, my apologies. I see now you are binding the context on the handler invocation here.

Just trying to think from an outside perspective, would this not be confusing for someone attempting to implement custom behavior?

For example, if I want to call an instance method in my custom ENTER handler, I may write something like this

```jsx
const handlers = {
  Enter (event) {
    this.handleEnter()
  }
}

<Downshift keyDownHandlers={handlers} />
```

Where `handleEnter` is an instance method on my component. This code will break however, because from the internally bound context, `this` will not have any such `handleEnter` method.

Reply:
Maybe instead of binding things to `this`, we should just pass it as a second argument? I think using `this` could really be misleading...